python/ray/train/trainer.py

```diff
@@ -284,17 +264,12 @@ def _finish_training(self):
             A list of return values from calling ``train_func`` on each worker.
                 Each item corresponds to the return value from a single worker.
         """
-
-        self._backend_executor.pause_reporting()
-        # Finish up processing checkpoints. Reporting has been disabled.
-        # Results will not be processed.
-        self._finish_checkpointing()
```

Comment:
Change 2: Remove this logic in `_finish_training`.

The session now reports the metrics + checkpoint in a single step. `session.get_next() -> (metrics, checkpoint)`

So, there's no longer 2 types of "training results" that need to be handled separately.

Reply:
The only code that runs `_finish_training` is:

https://github.com/ray-project/ray/blob/8d52e0c1d9ac5891abf975d9b59aed1d74050669/python/ray/train/trainer.py#L161-L165

`session.get_next()` -> `backend_executor.get_next_results()` -> `TrainingIterator._fetch_next_result` returns None when ALL training workers have exited the loop.

This means there cannot be any more `train.report` calls that need to be handled.